controllers/resource/reconciler_test.go

```diff
@@ -248,6 +252,21 @@ func TestClusterReconcilerReconcileVSphere(t *testing.T) {
 							t.Errorf("unmarshal failed: %v", err)
 						}
 						assert.Equal(t, expectedMCDeployment, template, "values", expectedMCDeployment, template)
+					case "KubeadmControlPlane":
+						kubeadmControlPlane := &controlplanev1.KubeadmControlPlane{}
+						assert.NoError(t, runtime.DefaultUnstructuredConverter.FromUnstructured(template.Object, kubeadmControlPlane))
```

Comment:
You might consider using `require.NoError` here, as that will end the test, preventing the user from having a slew of errors that aren't really errors, they just all failed because this failed.

Reply:
Alternatively, one can do:

```go
if assert.NoError(t, blah) {
  assert.Equal(t, strippedSSHKey, ...)
}
```

But a lot of people understandably don't like having branching logic (even simple branching logic) in their test checks.